Fix EGL support on macOS

mohd-akram opened the pull request:
This is a simpler version of #239 and is the patch we're using in MacPorts.

osy commented:
@ebassi Any chance this one can be reviewed? EGL support for macOS in QEMU is blocked on libepoxy support

ebassi commented:
The commit message is very underwhelming.

Why does removing the symbols from `dispatch_common.h` and changing the shared libraries in `dispatch_common.c` "fixes" macOS?

ebassi commented on src/dispatch_common.h:
```diff
@@ -27,10 +27,6 @@
 #define PLATFORM_HAS_EGL ENABLE_EGL
 #define PLATFORM_HAS_GLX ENABLE_GLX
 #define PLATFORM_HAS_WGL 1
-#elif defined(__APPLE__)
```
I'd rather keep this section, but change the values, instead of removing it and relying on the defaults. The defaults are generally understood to apply to Linux.

mohd-akram committed:
Fix EGL support on macOS

EGL is available on macOS via ANGLE. Use the dylibs provided by it and
ensure EGL is enabled in the code when it is in the build.
